specifications/datastore-environment-manager.org

```diff
@@ -0,0 +1,123 @@
+| field              | type   | description                                                                                  | example                                             |
+|--------------------+--------+----------------------------------------------------------------------------------------------+-----------------------------------------------------|
+| provider           | string | Name of build provider, e.g. ["host", "lxd", "multipass", "launchapad"]                      | "multipass"                                         |
+| timestamp_created  | string | Creation timestamp of build environment instance using python's =time.asctime()=.            | "Fri Aug 21 15:56:20 2020"                          |
+| timestamp_accessed | string | Accessed (last use) timestamp of build environment instance using python's =time.asctime()=. | "Fri Aug 21 15:56:20 2020"                          |
+| snapcraft_version  | string | Version of snapcraft that created this environment.                                          | "4.0.1"                                             |
+| project_name       | string | Name of project (as derived from snapcraft.yaml).                                            | "my-snap-name"                                      |
+| project_path       | string | Path to project.                                                                             | "/home/user/git/my-snap-name"                       |
+| project_id         | string | Unique identifier for project.                                                               | "<project-id>" as defined in environment lifecycle. |
+|--------------------+--------+----------------------------------------------------------------------------------------------+-----------------------------------------------------|
+
+- Whenever an environment is created or updated, the record is created/updated.  When cleaning the environment, the =build-environment= record is purged.
+
+*** LXD
+
+The datastore will have a =LXD= table with an array of =lxd-environment= records:
+
+|--------------------+--------+----------------------------------------------------------------------------------------------+-----------------------------------------------------|
+| field              | type   | description                                                                                  | example                                             |
+|--------------------+--------+----------------------------------------------------------------------------------------------+-----------------------------------------------------|
+| project_id         | string | Unique identifier for project.                                                               | "<project-id>" as defined in environment lifecycle. |
+| image_name         | string | Image used to create project.                                                                | "ubuntu:20.04"                                      |
```

Comment:
We can record that as well.  It can also be queried from the instance.

The purpose for this was for comparison checks to ensure `image_name` that Snapcraft wants is the same... the fingerprint doesn't necessarily have to match, unless we want to force clean/refresh when upstream image changes?  Maybe the switchover to `releases` helps here...

Reply:
image_source
image_hash (for multipass)
image_fingerprint (for lxd)